api/envoy/extensions/credentials/oauth2/v3/oauth2.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.extensions.credentials.oauth2.v3;
+
+import "envoy/config/core/v3/http_uri.proto";
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.credentials.oauth2.v3";
+option java_outer_classname = "Oauth2Proto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/credentials/oauth2/v3;oauth2v3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: OAuth2 Credential]
+// [#not-implemented-hide:]
+// [#extension: envoy.credentials.oauth2]
+
+// OAuth2 Credential to inject into the proxied requests.
+//
+message OAuth2 {
```

Comment:
Does this require the [oauth2](https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/http/oauth2/v3/oauth.proto) filter to be added prior to the injector? If not, can you explain why?
What I'm trying to avoid is adding different features in the 2 APIs.

Reply:
oauth2 extension is used to get token via oauth2 client credentials grant and inject it into the http header. It has no overlapping with the existing oauth2 filter.